.github/workflows/ci_cmake.yml

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) 2023-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+name: CI_CMAKE
+on: [push, pull_request]
+
+concurrency:
+  group: ci-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  CI_FAILFAST_TEST_LEAVE_DANGLING: 1  # GHA does not care about dangling processes and setting this variable avoids killing the CI script itself on error
+  MAKEJOBS: '-j10'
+
+jobs:
+  linux:
+    name: 'test linux'
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          submodules: recursive
+
+      - name: Setup
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y clang ccache build-essential cmake pkgconf python3-zmq libevent-dev bsdmainutils libboost-filesystem-dev libboost-test-dev libdb5.3++-dev libminiupnpc-dev libzmq3-dev libqrencode-dev systemtap-sdt-dev bpfcc-tools bpftrace
+
+      - name: Enable core dumps
+        run: ulimit -c unlimited
+
+      - name: Compile and run tests
+        run: |
+          CC=clang CXX=clang++ cmake -B build -DWERROR=ON -DWITH_ZMQ=ON -DBUILD_GUI=ON -DBUILD_BENCH=ON -DBUILD_FUZZ_BINARY=ON -DWITH_BDB=ON -DWITH_USDT=ON -DCMAKE_CXX_FLAGS='-Wno-error=unused-member-function'
+          cmake --build build -j $(nproc)
+          export CTEST_OUTPUT_ON_FAILURE=ON
+          echo "---------------- Build unit test ---------------- "
+          cmake --build build --target test_tapyrus -j $(nproc)
+          echo "---------------- Start unit test ---------------- "
+          cd ${{ github.workspace }}/build/src/test
+          ctest --output-on-failure -j 1 --timeout 300
+          echo "----------------  Start util test ---------------- "
+          cd ${{ github.workspace }}
+          cmake --build build --target check-tapyrus-util -j $(nproc)
+          echo "----------------  Start functional test ---------------- "
+          cd ${{ github.workspace }}
+          ${{ github.workspace }}/test/functional/test_runner.py -j $(( $(nproc) * 2 )) --combinedlogslen=99999999 --coverage --failfast --quiet 
+          echo "---------------- DONE ---------------------"
+
+      - name: Debug Core Dump (if exists)
+        if: failure()
+        run: |
+          if [ -f core ]; then
+            gdb -batch -ex "bt" -ex "quit" ./test_binary core
+          fi
+
```

Comment:
For Linux, just like with current CI, testing is required on both ARM and x86 architectures.